docs/website/docs/developers/debugging/gpu.md

```diff
@@ -236,3 +251,22 @@ useful approachs and tips:
     * [:simple-vulkan:] Use `--vulkan_robust_buffer_access=true` to `iree-run-module`
       especially when seeing undeterministic/corrupted contents in buffers and
       suspecting there are buffer allocation/indexing issues.
+
+## Binary substiution for ROCm
+
+[:simple-amd:] The AMD ROCm target supports binary substitution on HSA code objects
+(`.hsaco` files).
+
+These files are, under the hood, ELF shared libraries containing kernel code.
+
+If you have manually produced a binary you want to test, such as by manually
+running `llc` with different optiumization flags, you can turn the `.o` into
+a `.hsaco` with
+
+``` shell
+ld.lld -o [filename].hsaco -shared [filename].o
+```
+
+You'll want to use the LLD shipped by ROCm (typically in `/opt/rocm/llvm/bin`)
```

Comment:
-1, don't advertise this, the compiler uses one in the llvm dir and that's what we should be using